### DISCUSSION TITLE
Split NextJS static build process : it runs too long!

### DISCUSSION THREAD

**Discussion by lilithebowman** [vercel/next.js#26451] [Ideas]
### Describe the feature you'd like to request

Currently on our local machines and on our UAT environments the Next JS build step takes over an hour.

We have about 265 pages right now and more are coming. Most of them are document display pages which are sub sections of sub sections.

Our build process currently fails almost every time. We keep extending the build time available on AWS and it keeps finding new time limits to exceed.

It's not actually that big of a site.

### Describe the solution you'd like

A way to chunk the build process so that we can manage which parts of the process are taking longer and plan / fix accordingly.

This would have the added benefit of perhaps making us only redo longer query builds instead of redoing the whole build every time something fails in the build step.

### Describe alternatives you've considered

Not using Next JS (not actually a possibility at this point considering how deep we are in this project cycle.

**Comment by awareness481** [comment]
I'm not sure what's going on but that doesn't seem normal. Currently working on a project with around 2k pages I don't think it takes longer than 5 mins to build.

This makes it sound like there's some underlying issues on your project which stall page creation.

**Comment by lilithebowman** [comment]
NextJS > Apollo > GraphQL > WordPress is the stack.

AWS machines are much lower clock speed than your local machine is likely to be. They're server processors with many slow cores. That could be one problem.

Each page calls several components all of which have a GraphQL query to Wordpress for info. Maybe that's the slower part from what you're used to.

This is a pretty common use case. Common enough for there to be some momentum behind it. So I think this feature would likely help someone other than myself.

🤷‍♀️
